source/dub/internal/vibecompat/inet/url.d

```diff
@@ -215,16 +210,10 @@ struct URL {
 		auto dst = appender!string();
 		dst.put(schema);
 		dst.put(":");
-		switch(schema){
-			default: break;
-			case "file":
-			case "http":
-			case "https":
-			case "ftp":
-			case "spdy":
-			case "sftp":
-				dst.put("//");
-				break;
+		auto schema_parts = schema.split("+");
+		if (schema_parts.any!(p => p.canFind("http", "https", "ftp", "spdy", "file", "sftp")))
```

Comment:
Shouldn't this be just `.back`?

Reply:
I do not understand how the range .back function is related to this source code part. Could you explain what you mean?

Reply:
```diff
diff --git a/source/dub/internal/vibecompat/inet/url.d b/source/dub/internal/vibecompat/inet/url.d
index 54c3699..d2c346a 100644
--- a/source/dub/internal/vibecompat/inet/url.d
+++ b/source/dub/internal/vibecompat/inet/url.d
@@ -211,7 +211,7 @@ struct URL {
 		dst.put(schema);
 		dst.put(":");
 		auto schema_parts = schema.split("+");
-		if (schema_parts.any!(p => p.canFind("http", "https", "ftp", "spdy", "file", "sftp")))
+		if (schema_parts.back.canFind("http", "https", "ftp", "spdy", "file", "sftp"))
 		{
 			dst.put("//");
 		}
```

Reply:
back could RangeError if schema_parts is empty

Reply:
@wilzbach I included your suggestions. Thanks.
@WebFreak001 Range error is checked using "empty" function